utils/json-utils/src/test/scala/pl/touk/nussknacker/engine/json/SwaggerBasedJsonSchemaTypeDefinitionExtractorTest.scala

```diff
@@ -511,4 +511,56 @@ class SwaggerBasedJsonSchemaTypeDefinitionExtractorTest extends AnyFunSuite with
     result shouldBe TypedObjectTypingResult(List("items" -> TypedObjectTypingResult(List("next" -> Unknown, "value" -> Typed[String]))))
 
   }
+
+  test("should map integer type to the narrowest type possible") {
+    def schema(minValue: String, exclusiveMinValue: String, maxValue: String, exclusiveMaxValue: String) = JsonSchemaBuilder.parseSchema(
+      s"""{
+        |   "type":"object",
+        |   "properties":{
+        |      "id":{
+        |        "type": "integer"
+        |        ${Option(minValue).map(min => s""", "minimum":$min""").getOrElse("")}
+        |        ${Option(exclusiveMinValue).map(min => s""", "exclusiveMinimum":$min""").getOrElse("")}
+        |        ${Option(maxValue).map(max => s""", "maximum":$max""").getOrElse("")}
+        |        ${Option(exclusiveMaxValue).map(max => s""", "exclusiveMaximum":$max""").getOrElse("")}
+        |       }
+        |   }
+        |}""".stripMargin)
+
+    val table = Table(
+      ("minValue", "exclusiveMinValue", "maxValue", "exclusiveMaxValue", "expectedType"),
```

Comment:
Can you add tests for the same minValue and maxValue?

Reply:
done